### PR TITLE
Fix nodoc to internal class error document some of them

### PR DESCRIPTION
**Pull request by vipulnsward**
[ci skip]

Continuation of and Fixes #20808

### CODE REVIEW

File: actionpack/lib/action_controller/metal/head.rb [hunk 1]
```diff
@@ -50,16 +50,16 @@ def head(status, options = {})
...
     private
-    # :nodoc:
-    def include_content?(status)
-      case status
-      when 100..199
-        false
-      when 204, 205, 304
-        false
-      else
-        true
```
Comment (prathamesh-sonpatki) [inline actionpack/lib/action_controller/metal/head.rb]: @vipulnsward Can you keep indentation as it is and move `nodoc` at correct place?

File: activesupport/lib/active_support/core_ext/string/output_safety.rb [hunk 2]
```diff
@@ -141,6 +141,7 @@ class SafeBuffer < String
     alias_method :original_concat, :concat
     private :original_concat
 
+    # Raised when <tt>ActiveSupport::SafeBuffer#safe_concat</tt> is called on unsafe buffers
```
Comment (prathamesh-sonpatki) [inline activesupport/lib/active_support/core_ext/string/output_safety.rb]: Add `.` at the end.

**Comment by prathamesh-sonpatki** [comment]
@vipulnsward some of the changes are overlapping with https://github.com/rails/rails/pull/20808, can you pull in those changes here and give credit to @TheBlasfem.

File: actionpack/lib/action_controller/metal/head.rb (hunk 1 above)
Comment (vipulnsward) [inline actionpack/lib/action_controller/metal/head.rb]: @prathamesh-sonpatki this change is intentional.

**Comment by vipulnsward** [comment]
Yes, I forgot to do that.

**Comment by vipulnsward** [comment]
Updated.

File: actionpack/lib/action_controller/metal/head.rb (hunk 1 above)
Comment (prathamesh-sonpatki) [inline actionpack/lib/action_controller/metal/head.rb]: We try to keep indentation as it is. Any particular reason for this change?

Comment (vipulnsward) [inline actionpack/lib/action_controller/metal/head.rb]: Was following convention. But yes, makes sense to keep as is.

File: actionpack/lib/action_controller/metal/head.rb [hunk 3]
```diff
@@ -50,8 +50,8 @@ def head(status, options = {})
     end
 
     private
-    # :nodoc:
-    def include_content?(status)
+
+    def include_content?(status) # :nodoc:
```
Comment (maclover7) [inline actionpack/lib/action_controller/metal/head.rb]: Do we really need to mark the method with `nodooc`, even though the method' private? I believe private methods do not show up in documentation.

**Comment by maclover7** [comment]
One small comment, then ready to merge.

r? @maclover7

**Comment by kaspth** [comment]
@maclover7 Vipul is on the issues team and can merge doc things himself ;)

File: actionpack/lib/action_controller/metal/head.rb [hunk 4]
```diff
@@ -50,7 +50,6 @@ def head(status, options = {})
     end
 
     private
-    # :nodoc:
     def include_content?(status)
```
Comment (vipulnsward) [inline actionpack/lib/action_controller/metal/head.rb]: Verified, we don't need this. Output without it- http://take.ms/EMSJM

**Comment by maclover7** [comment]
:+1: Thank you for your contribution!

**Comment by maclover7** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**